include/soralog/macro.hpp

```diff
@@ -21,13 +21,13 @@
  */
 
 #define _SL_LOG_IF_LEVEL(LOG, LVL, FMT, ...)                 \
-  do {                                                       \
+  ({                                                         \
```

Comment:
Do we allow/use https://gcc.gnu.org/onlinedocs/gcc/Statement-Exprs.html ?

If we use it, then we may use [`BOOST_OUTCOME_TRYX`](https://live.boost.org/doc/libs/1_83_0/libs/outcome/doc/html/tutorial/essential/result/try.html#compiler-specific-extension-boost-outcome-tryx)
```
// OUTCOME_TRY but it's expression
auto x = BOOST_OUTCOME_TRYX(foo()) + BOOST_OUTCOME_TRYX(bar());

#define BOOST_OUTCOME_TRYX(expr)
  ({
    auto tmp = expr;
    // return inside statement expression
    if (tmp.has_error()) return error;
    tmp.value();
  })
```

Reply:
Why not? This is topic for discuss